### PR TITLE
feat: allow smart update of container exec and spark configs

### PR DESCRIPTION
**Pull request by aardon-dku**

### CODE REVIEW

**Review by duckie — CHANGES_REQUESTED** [review]
There was a misunderstanding of the spec, lets re-work as discussed

**Review by duckie — CHANGES_REQUESTED** [review]
Let the smart update be the default but offer a way to disable it.

File: plugins/modules/dss_general_settings.py [hunk 1]
```diff
@@ -119,21 +130,34 @@
...
+    }
+}
 
 encrypted_fields = [
     "ldapSettings.bindPassword", "ssoSettings.samlSPParams.keystorePassword", "ssoSettings.openIDParams.clientSecret",
     "azureADSettings.credentialsClientSecret", "azureADSettings.credentialsCertificatePassword"
 ]
+smart_update_fields = [
+    "containerSettings.executionConfigs", "sparkSettings.executionConfigs"
+]
```
Comment (duckie) [inline plugins/modules/dss_general_settings.py:151]: Could be built from `smart_update_fields_template`

**Review by duckie — APPROVED** [review]

File: plugins/modules/dss_general_settings.py (hunk 1 above)
Comment (aardon-dku) [reply plugins/modules/dss_general_settings.py:143]: refactored in [0bfb489](https://github.com/dataiku/dataiku-ansible-collection/pull/5/commits/0bfb489462c9ff843fd81576dc7c039c21f5eead)[0bfb489](https://github.com/dataiku/dataiku-ansible-collection/pull/5/commits/0bfb489462c9ff843fd81576dc7c039c21f5eead)

**Pull request merged**